libsql-sqlite3/test/libsql_vector_index.test

```diff
@@ -307,6 +321,8 @@ do_execsql_test vector-f64-index {
   SELECT * FROM vector_top_k('t_f64_idx', vector64('[1,2]'), 2);
 } {1 2}
 
+reset_db
```

Comment:
We currently have some issue and test without db reset starting to generate generic sqlite error in some unexpected places.

This is really weird but I don't understand why it happens and how we can fix it.
Last time I investigated this issue I noticed that `SELECT` over `libsql_vector_meta_shadow` table with condition over `name` field sometimes returns nothing but if you will force full scan of the table - then everything will be fine.

I suspect that issue arise when split of `libsql_vector_meta_shadow` B-Tree happens. But still not sure why we have this issue...

CC @penberg